pkg/injector/envoy_container.go

```diff
@@ -15,21 +15,46 @@ const (
 	envoyProxyConfigPath     = "/etc/envoy"
 )
 
-func getEnvoySidecarContainerSpec(pod *corev1.Pod, cfg configurator.Configurator, originalHealthProbes healthProbes) corev1.Container {
+func getUnixSecurityContext() *corev1.SecurityContext {
+	return &corev1.SecurityContext{
+		RunAsUser: func() *int64 {
+			uid := constants.EnvoyUID
+			return &uid
+		}(),
+	}
+}
+
+func getWindowsSecurityContext() *corev1.SecurityContext {
+	return &corev1.SecurityContext{
+		WindowsOptions: &corev1.WindowsSecurityContextOptions{
+			RunAsUserName: func() *string {
+				userName := "EnvoyUser"
+				return &userName
+			}(),
+		},
+	}
+}
+
+func getEnvoySidecarContainerSpec(pod *corev1.Pod, cfg configurator.Configurator, originalHealthProbes healthProbes, isWindowsPod bool) corev1.Container {
 	// cluster ID will be used as an identifier to the tracing sink
 	clusterID := fmt.Sprintf("%s.%s", pod.Spec.ServiceAccountName, pod.Namespace)
 
+	var securityContext *corev1.SecurityContext
+	var containerImage string
+	if isWindowsPod {
```

Comment:
Can you consolidate these into one method that takes the OS type as a param?

Reply:
Yeah you are right, havent fully internalized that golang has function with multiple return variables

Reply:
Instead of passing `isWindows` bool around, let's directly pass `osType` as a string argument and have helper functions use this for appropriate config.

Reply:
fixed